src/mavedb/scripts/export_public_data.py

```diff
@@ -0,0 +1,101 @@
+"""
+Script that generates a dump of published MaveDB data.
+
+Usage:
+```
+python3 -m mavedb.scripts.export_public_data
+```
+
+This generates a ZIP archive named `mavedb-dump.zip` in the working directory. the ZIP file has the following contents:
+- main.json: A JSON file providing metadata for all of the published experiment sets, experiments, and score sets
+- variants/
+  - [URN].counts.csv (for each variant URN): The score set's variant count columns,
+    sorted by variant number
+  - [URN].scores.csv (for each variant URN): The score set's variant count columns,
+    sorted by variant number
+
+In the exported JSON metadata, the root object's `experimentSets` property gives an array of experiment sets.
+Experiments are nested in their parent experiment sets, and score sets in their parent experiments.
+
+The variant URNs used in filenames do not include the `urn:mavedb:` scheme identifier, so they look like
+`00000001-a-1.counts.csv` and `00000001-a-1.scores.csv`, for instance.
+
+Unpublished data are not included in the dump, and user details are limited to ORCID IDs and names of contributors to
+published data sets.
+"""
+
+from datetime import datetime, timezone
+from itertools import chain
+import json
+import logging
+from typing import Callable, Iterable, TypeVar
+from zipfile import ZipFile
+
+from fastapi.encoders import jsonable_encoder
+from sqlalchemy import select
+from sqlalchemy.orm import lazyload
+
+from mavedb.lib.score_sets import get_score_set_counts_as_csv, get_score_set_scores_as_csv
+from mavedb.lib.script_environment import init_script_environment
+from mavedb.models.experiment import Experiment
+from mavedb.models.experiment_set import ExperimentSet
+from mavedb.models.score_set import ScoreSet
+from mavedb.view_models.experiment_set import ExperimentSetPublicDump
+
+db = init_script_environment()
+
+logger = logging.getLogger(__name__)
+
+S = TypeVar("S")
+T = TypeVar("T")
+
+
+def flatmap(f: Callable[[S], Iterable[T]], items: Iterable[S]) -> Iterable[T]:
+    return chain.from_iterable(map(f, items))
+
+
+experiment_sets_query = db.scalars(
+    select(ExperimentSet)
+    .where(ExperimentSet.published_date.is_not(None))
+    .options(
+        lazyload(ExperimentSet.experiments.and_(Experiment.published_date.is_not(None))).options(
+            lazyload(Experiment.score_sets.and_(ScoreSet.published_date.is_not(None)))
```

Comment:
I see you made processing state part of the export - I think we should also not export unprocessed score sets. it might violate user expectations re: the completeness of score sets in the export (they won't have corresponding scores files, for example).

Reply:
My assumption is that unprocessed score sets aren't published. Is that valid? Maybe it's possible that a published score set gets updated with new data, but I don't think we plan to allow this except by administrative intervention.

I considered excluding processing state from the JSON metadata, but figured I'd leave it in for consistency with the API.